jenkins/java/ql/src/WebMethodMissingPermissionCheck.md

```diff
@@ -1,9 +1,75 @@
+3. If this is a true positive finding, use the documentation below to resolve it.
+
+If this finding is in a plugin hosted by the Jenkins project, you can also always [contact the Jenkins Security Team via Jira or email](https://www.jenkins.io/security/#reporting-vulnerabilities) to ask for help in resolving this finding.
+
+## Is this a false positive?
+
+This check can easily identify _false positives_, i.e. safe code that is marked as being unsafe, for any of the following reasons:
+
+1. The information being returned is not private.
+2. The method does not have side effects.
+3. The method is otherwise protected.
+4. The method is not actually routable.
+
+### What is private information?
+
+Anything a user would not usually be entitled to have access to without additional permissions:
+
+1. Information about the configuration of the system usually requires Overall/Manage, Overall/SystemRead, or Overall/Administer permission.
+2. Information about the configuration of a job usually requires Job/Configure or Job/ExtendedRead permission (although some information will be displayed to users with Job/Read permission -- and that information does not need further permission to access).
+
+As a rule of thumb, if a user with a given level of access can obtain the information through an intended UI or API, it's not a problem.
+It's also not a problem if a user could obtain the same information by reading the source code of the plugin (e.g. `doFill` methods providing a fixed list of combobox options).
+It's only a problem when direct browsing to URLs would reveal information they would not otherwise be able to access.
+
+### What are side effects?
+
+Side effects for the purpose of this check are any of of the following:
+
+1. Changing system state (e.g. a URL handling form submissions will change some configuration).
+2. Connecting to other services, e.g. by sending HTTP requests. A notable exception to this is plugins integrating with [Credentials Plugin](https://plugins.jenkins.io/credentials/): While a permission check _is_ needed (see [the Credentials Plugin consumer guide](https://github.com/jenkinsci/credentials-plugin/blob/master/docs/consumer.adoc#providing-a-ui-form-element-to-let-a-user-select-credentials)), it's because enumerating credentials is restricted to users with the appropriate permissions, rather than the potential side effect of accessing [`CredentialsProvider`](https://www.jenkins.io/doc/developer/extensions/credentials/#credentialsprovider) over the network.
+3. Expensive computations, especially if the cost is depending on user input.
+
+### Other methods of protection
+
+Stapler request routing does not require that each individual web method perform a permission check. A common solution if entire objects are intended to be protected from lower-privilege users is to implement `StaplerProxy` and use the `#getTarget` method to implement a permission check, returning `this` when it is successful.
```

Comment:
Proposing to add commas to make it more readable: "A common solution, if entire objects are intended to be protected from lower-privilege users, is to implement `StaplerProxy` and use the `#getTarget` method to implement a permission check, returning `this` when it is successful."

Reply:
```suggestion
Stapler request routing does not require that each individual web method perform a permission check. A common solution, if entire objects are intended to be protected from lower-privilege users, is to implement `StaplerProxy` and use the `#getTarget` method to implement a permission check, returning `this` when it is successful.
```